### ISSUE TITLE
azurerm_mssql_database: valid max_size_gb reported as invalid

### ISSUE THREAD

**Issue by valleypontiff** [hashicorp/terraform-provider-azurerm#18374]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.9

### AzureRM Provider Version

3.22.0

### Affected Resource(s)/Data Source(s)

azurerm_mssql_database

### Terraform Configuration Files

```hcl
resource "azurerm_mssql_database" "sqldb1" {
  name        = "sqldb-t1-dpsb1"
  server_id   = azurerm_mssql_server.sql.id
  max_size_gb = 10
  sku_name    = "S1"
  create_mode = "Default"
}
```

### Debug Output/Panic Output

```shell
Error: creating/updating Database: (Name "sqldb-t1-dpsb1-dev" / Server Name "sql-t1-dpsb1-dev" / Resource Group "rgx"): sql.DatabasesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidMaxSizeTierCombination" Message="The tier 'Standard' does not support the database max size '-2147483648'."
```

### Expected Behaviour

The size is accepted and the database is created with a max size of 10 GB.

### Actual Behaviour

Refuses to create the database (seemingly due to an error in conversion of GB to B). (When max_size_gb is 11, it reports the max size as '-1073741824'; when max_size_gb is 9, it creates the database with an actual max size of 1 GB.)

### References

#8890

**Comment by sinbai** [comment]
@valleypontiff thanks for raising this issue here. Could you provide your go version like:
![image](https://user-images.githubusercontent.com/39109137/190347435-16e100e4-53c1-4929-891c-cb0e4dfd5563.png)

Also, It is best to provide the operation system type in Device specifications, like:
```
Device name	XXXX
Processor	Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz   2.90 GHz
Installed RAM	16.0 GB (15.7 GB usable)
Device ID	XXXX
Product ID	XXXX
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display
```

**Comment by valleypontiff** [comment]
go is not installed.

Processor	Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz
Installed RAM	16.0 GB (15.7 GB usable)
System type	64-bit operating system, x64-based processor

**Comment by sinbai** [comment]
@valleypontiff Thanks for your quick reply. So, could you share the details of terraform version, like:
![image](https://user-images.githubusercontent.com/39109137/190538421-1dc85983-e8a3-43bc-917e-21f8a0c5e63f.png)

**Comment by valleypontiff** [comment]
Terraform v1.2.9
on windows_386
+ provider registry.terraform.io/hashicorp/azuread v2.28.1
+ provider registry.terraform.io/hashicorp/azurerm v3.22.0
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.4.3
+ provider registry.terraform.io/hashicorp/time v0.8.0

**Comment by CSanches** [comment]
I'm facing the same issue. 
I had tried 3.49, 3.48 and 3.40 providers.
If I set to create a S1 DB with 200GB (less than the included), it will create with 250GB ignoring my initial size.
If I set to create a S1 DB with 250GB (the same as included), it creates the error message:
Message=\"The tier 'Standard' does not support the database max size '-2147483648'
If I remove the "max_size_gb" from the TF file, it creates with the included value (250GB).
